src/main/java/hudson/plugins/git/UserRemoteConfig.java

```diff
@@ -146,12 +147,18 @@ public FormValidation doCheckUrl(@AncestorInPath Job project,
                 return FormValidation.ok();
 
             // get git executable on master
-            final EnvVars environment = project.getEnvironment(Jenkins.getActiveInstance(), TaskListener.NULL);
+            EnvVars environment;
+            final Jenkins jenkins = Jenkins.getActiveInstance();
+            if (item instanceof Job) {
+                environment = ((Job) item).getEnvironment(jenkins, TaskListener.NULL);
+            } else {
+                environment = jenkins.getComputer("(master)").buildEnvironment(TaskListener.NULL);
```

Comment:
`jenkins.getComputer("(master)")`

WAT!!!

Isn't that just `Jenkins.getActiveInstance().toComputer()` seems wrong that we are looking up based on a **displayName**